client/signup/jetpack-connect/authorize-form.jsx

```diff
@@ -106,7 +106,7 @@ const LoggedInForm = React.createClass( {
 	componentWillMount() {
 		const { autoAuthorize, queryObject } = this.props.jetpackConnectAuthorize;
 		debug( 'Checking for auto-auth on mount', autoAuthorize );
-		if ( autoAuthorize ) {
+		if ( autoAuthorize || this.isCalypsoStartedConnection() ) {
```

Comment:
`autoAuthorize` gets set if a user is creating a new WPCOM account. In that case, they'll see a 'Sign Up and Connect' button. After sign up, a `JETPACK_CONNECT_CREATE_ACCOUNT` action will be dispatched which will add `autoAuthorize` to the redux state. This will allow the authorization to kick off without the need for an additional click. I think your implementation is solid.